source/include/StringUtilities.h

```diff
@@ -424,6 +427,42 @@ class DecodedSubString {
     Utf8Char *_decodedStr;
 };
 
+//------------------------------------------------------------
+//! A class for making temporary %-code encoded SubStrings
```

Comment:
Is this %-code encoding something unique to us or is it something that is more common in the industry? If the answer is the industry, then we should use the name used in the industry to describe this algorithm and have more test cases to ensure we are properly implementing it.